projects/igniteui-angular/src/lib/grids/headers/grid-header-row.component.ts

```diff
@@ -211,9 +211,9 @@ export class IgxGridHeaderRowComponent implements DoCheck {
         }
 
         if (this.grid.selectionService.areAllRowSelected()) {
-            this.grid.selectionService.clearRowSelection(event);
+            this.grid.selectionService.clearRowSelection(event, false);
```

Comment:
It seems obvious that on clearRowSelection, isSelectAll would be false. Maybe this can be omitted.